### ISSUE TITLE
VoiceOver does not read the tree views properly in newer versions of TeamTalk

### ISSUE THREAD

**Issue by GruiaChiscop** [BearWare/TeamTalk5#2421]
### Description

When I'm focused on a tree view, such as channel list or server list on teamTalk and try to read its content voiceOver doesn't give any feedback, sometimes it reads some items from the tree view, but not the ones that are focused.

### Application

- [X] qtTeamTalk
- [ ] TeamTalkAndroid
- [ ] iTeamTalk
- [ ] TeamTalkClassic
- [ ] TeamTalkServer

### Platform

- [ ] Windows
- [X] macOS
- [ ] Android
- [ ] iOS
- [ ] Linux

### Expected behavior

VoiceOver should read the focused item in the tree views when moving through them.

### Actual behavior

When moving through the tree view items, voiceOver doesn't speak anything or, sometimes, it speaks the previous focused items in the tree view, but not the actual focused ones.

### Steps to reproduce problem

Try to log in to a server with more than one channel or users logged on, focus the channel list tree view and move through each items in it. You should see how VoiceOver behaves.

**Comment by Muamalaljanahi** [comment]
I have the same issue, that's why I am using iTeamTalk the App Store version

**Comment by Nick6489** [comment]
This has gotten worse - a lot worse - since QT 6.10 Beta. Sadly, updating to QT 6.10 beta was necessary in order to close #2472. Now, in the TeamTalk 5.19 beta release, *nothing* reads in the channel table, or the list of saved servers. We're back to where things used to be before QT grew accessibility at all.

**Comment by vlad-a-c** [comment]
hello. yes, i can confirm that. server list isn't readable at all. i basically either need to use ocr or randomly go on servers.

**Comment by vlad-a-c** [comment]
ok,i came up with an idea. i know other programmers that use qt or something similar which is very inaccessible at least on mac side. basically for fixing the problem you could make the tts engine that reads the speech events to also read the tree views and also anounce what's expanded or not.
it's not the best solution, but until the QT devs do not fix the accessibility issues you could make this work around that could work for people.

**Comment by bear101** [comment]
@CoBC didn't we once have a feature where Preferences -> Text to Speech had an option to TTS items in the tree view?

**Comment by CoBC** [comment]
Yes, we had it. I removed it when Qt fixed component accessibility. 
Should I reintroduce it?

**Comment by bear101** [comment]
Yes, I think so. It looks like there's not coming a solution any time soon.

**Comment by vlad-a-c** [comment]
they fixed the accessibility component only on windows. on mac it's very broken.

**Comment by bear101** [comment]
Can you try this version and see if it fixes the problem with VoiceOver and the tree control:
https://bearware.dk/beta/TeamTalk_v5.19_Qt6.9.2_Setup.pkg

**Comment by Nick6489** [comment]
I just did. No change at all, it still functions the same way as before. No spoken feedback when arrowing through treviews.

**Comment by bear101** [comment]
When I use Control+Option+Down Arrow in the channels tree view then I can hear each item in the tree using VoiceOver.

**Comment by Nick6489** [comment]
[TeamTalk struggling with the UI.mp3](https://github.com/user-attachments/files/22013372/TeamTalk.struggling.with.the.UI.mp3)Attached me struggling with the UI. I did find some things that are better than they were, but what I've discovered is that repeatability isn't really a thing with this interface. It never really has been, but it's gotten a lot worse since 2023.

**Comment by bear101** [comment]
@CoBC can you look into reenabling the feature where tree items were spoken with TTS? Maybe it's just to revert a commit.

**Comment by CoBC** [comment]
Yes, I'll do it

**Comment by bear101** [comment]
In this version we reintroduced the option to make TTS announce table item changes: https://bearware.dk/beta/TeamTalk_v5.20.0.5173_Setup.pkg

**Comment by vlad-a-c** [comment]
ok, but now e have another issue with this. on this newer version you gave us we can't ctrl click on the right server for editing it's details. it selects some random one from the list. i didn't actually manage to figure a work around to select one. something happened to the server list box, cause the server doesn't say selected.

**Comment by Nick6489** [comment]
I hate to resurrect this, but it appears we're back to where we were in new installations of the 5.23 beta. I can't find where you're supposed to set it so you can have your TTS backend speak the items in the treview. Oddly, changing from QT Accessibility to Prism broke it for me, and changing it back didn't fix it. What's the state of this feature?
